docs/architecture/decisions/0002-convert-to-static.md

```diff
@@ -0,0 +1,104 @@
+ - Reduce hosting costs (k8s clusters, ELBs, databases)
+ - Reduce code maintenance cost (urgent python upgrades / security
+   vulnerabilities) and overall engineering time.
+
+#### What enables us to consider this option?
+
+ - Historically Careers used to have a Admin interface to submit University
+   Days. This has been replaced by a static file named `university_events.yml`
+ - Also Careers was setup for multi-locale support but only en-US was ever
+   implemented. There 're no plans to support more locales in the future.
+ - There's currently no site functionality which requires a dynamic back-end.
+   The site can get rebuilt and pushed as static HTML whenever
+   `university_events.yml` changes or a Job Posting in Greenhouse gets created,
+   updated or deleted.
+
+### Build and Hosting Options
+
+We discussed a number of options for generating the static site including:
+
+ - K8s cron jobs
+ - Apache Airflow / Google Cloud Composer
+ - GitHub Actions:
+ - TravisCI
+ - AWS Lambda
+ - CircleCI
+ - GitlabCI
+ - Jenkins
+   - Would work but we want to move away from our current Jenkins instance.
+
+The static site would be hosted in S3 with CloudFront CDN in front.
+
+Another suggested option was [Netlify](http://netlify.com/) which can take care
+of the generation and the hosting of the website. Since Netlify has been brought
+up a few times as a possible solution for other MEAO projects, it makes a good
+candidate to try with a lower profile site (compared to our other properties)
+like careers.m.o.
+
+Netlify offers a compelling set of features, including WebHooks (for directly
+triggering the build from Greenhouse #344), custom HTTP Headers for certain
+paths (like the RSS feeds #360).
+
+The issue motivating this decision, and any context that influences or
+constrains the decision.
+
+### Other key points taken into consideration
+
+ - The Green House integration includes some parse logic to normalize the format
+   of the posted jobs which we'll have to re-write / integrate to any static
+   site generation tool
+ - The Django site works just fine and it's well maintained. It will be upgraded
+   with minimal effort to Django 2.2 soon (see #339) which is an LTS release and
+   will be supported until at least April 2022
+ - We have very limited engineering time to devote to this effort
+
+
+## Decision
+
+We agreed to:
+ - Continue maintain the Django project and upgrade to Django 2.2 LTS release.
+ - Create scripts to generate a static copy of the Django site, along with the
+   required build acceptance tests to verify that the builds are correct.
+ - Use [Netlify](http://netlify.com/) to host a generated static copy of the
+   Django project.
+ - Setup a webhook from Greenhouse to Netlify to update the site whenever a Job
+   Post gets updated.
+
+## Consequences
+
+### Risks
+
+- Site does not update when Greenhouse updates:
+
+  The site update will be triggered by a web-hook executed by Greenhouse of
```

Comment:
We could also configure a daily (or X times per day) build to cover any webhook notification failures.

Reply:
That's a good idea but from where will we trigger this?

Reply:
would a periodic build be more secure/less complex than a webhook?

Reply:
> would a periodic build be more secure/less complex than a webhook?

This was my initial plan but then I liked how GH re-tried failed webhooks and even with an exponential back-off.

The problem with cronjobs is that we need to figure out where we're going to run the cron daemon (in the broader sense). If Jenkins is going away and we want to reduce the number of things we host, I guess it means we need to find a Webhook SaaS plaftform with cronjob support. Zapier could be a solution.

...few minutes later...

I setup a Zapier account and connected it with Netlify to trigger builds daily. Zapier is rather expensive for teams (~$300/mo) for what we want to do, but we can just use one account to do this.

I suggest that we keep GreenHouse instant webhook builds and Zapier for daily builds. Thoughts?

Reply:
> but let's not go to far with them until we get some other things figured out I'd say.

+1

Reply:
Added a line that says that we'll setup a webhook for daily rebuilds and monitor with DMS.